### ISSUE TITLE
Consistent return type for feature assertions

### ISSUE THREAD

**Issue by robstoll** [robstoll/atrium#92]
*Platform* (JVM, JS and/or Android): all

## Code related feature
```
assert(listOf(1,2))
  .get (1){ ... } // subject still List<Int> here
  .returnValueOf (subject::get, 1) { ... } // subject is now Int
```
This should be consistent. IMO it makes sense not to narrow the subject if one provides sub assertions (an assertionCreator lambda) as we can continue using the fluent API.
One can use the overload without lambda if he wants to narrow the subject.

There is a side effect involved in using sub assertions as they form an assertion group meaning all sub assertions are evaluated, ergo no fail fast.
One can chain the assertions within the assertion block though. Following explained in code:
```
assert(listOf(1,5))
  .get(1) { 
    isLessThan(3) // fails
    isGreaterThan (1) // fails as well
  }
  .get(1) {
    isLessThan(3) //fails
       .and.isGreaterThan(1) // not evaluated
  }
```

**Comment by robstoll** [comment]
fixed by 5e229e0

**Comment by robstoll** [comment]
Not fully implemented yet, asIterable is returning Iterable in both cases

**Comment by robstoll** [comment]
rest implemented with 44b2c3a9
